src/pfs/pfs.go

```diff
@@ -343,6 +347,38 @@ func onlyHex(p []byte) bool {
 	return true
 }
 
+type ancestryRegexp struct {
+	*regexp.Regexp
+	partMap map[string][]byte
+}
+
+func makeAncestryRegexp(ancestryPart []byte) (ancestryRegexp, error) {
+	// ?P<> syntax assigns each capture group a name.
+	re := regexp.MustCompile(`^(?P<dot>\.\d+)?((?P<carrots>\^+)|(?P<carrotWithNumbers>\^\d+)|(?P<tildes>~+)|(?P<tildeWithNumbers>~\d+))?$`)
```

Comment:
I think you mean carets (^) rather than carrots (🥕 🥕 ).

Reply:
Today, I learned.